# HG changeset patch
# User andrew
# Date 1477589125 -3600
#      Thu Oct 27 18:25:25 2016 +0100
# Node ID fff29c78ee406a290e5b79dc14a1b1a98de1d0f4
# Parent  f23cc792224c21431b71da6a31ab751ac6f5c236
PR3209: Backed out changeset 3cc3ab869ccf

diff --git a/src/share/vm/opto/lcm.cpp b/src/share/vm/opto/lcm.cpp
--- openjdk/hotspot/src/share/vm/opto/lcm.cpp
+++ openjdk/hotspot/src/share/vm/opto/lcm.cpp
@@ -1068,14 +1068,11 @@
     Block *sb = _succs[i];
     // Clone the entire area; ignoring the edge fixup for now.
     for( uint j = end; j > beg; j-- ) {
+      // It is safe here to clone a node with anti_dependence
+      // since clones dominate on each path.
       Node *clone = _nodes[j-1]->clone();
       sb->_nodes.insert( 1, clone );
       bbs.map(clone->_idx,sb);
-#ifdef AARCH64
-      if (clone->needs_anti_dependence_check()) {
-        insert_anti_dependences(sb, clone);
-      }
-#endif
     }
   }
 
